dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/JarDumper.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.reflect.ClassPath.ClassInfo;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.bcel.Const;
+import org.apache.bcel.classfile.Attribute;
+import org.apache.bcel.classfile.ConstantPool;
+import org.apache.bcel.classfile.InnerClass;
+import org.apache.bcel.classfile.InnerClasses;
+import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.util.ClassPath;
+import org.apache.bcel.util.SyntheticRepository;
+
+/**
+ * This class reads a jar file and list method references in its class files
+ */
+class JarDumper {
+
+  /**
+   *  Lists all external method references from the jar file.
+   *
+   * @param jarFileUrl URL for the jar file
+   * @return list of the method signatures with their fully-qualified classes
+   * @throws IOException when there is a problem in reading the jar file
+   */
+  public static List<FullyQualifiedMethodSignature> listExternalMethodReferences(
+      URL jarFileUrl) throws IOException {
+    List<FullyQualifiedMethodSignature> methodReferences = new ArrayList<>();
+
+    SyntheticRepository repository = SyntheticRepository.getInstance(
+        new ClassPath(jarFileUrl.getFile()));
+    Set<String> internalClassNames = new HashSet<>();
+    for (ClassInfo classInfo : listTopLevelClassesFromJar(jarFileUrl)) {
+      String className = classInfo.getName();
+      JavaClass javaClass;
```

Comment:
don't split variable declaration from initialization. Instead put more code in the try block.

Reply:
Now the function has `throws`, there's no need to split declaration and initialization.